File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(long num) {
        int answer = 0;

        int count = 0;
        while (count<500) {

            if (num==1){
                break;
            }
            else if (num % 2 == 0) {
                num = num / 2;
            } else {
                num = num * 3 + 1;
            }
            count += 1;
            answer += 1;

        }
        if (num==1) return answer;
        else if (count>=500&&num!=1){
            answer = -1;
        }
        return answer;
    }
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.swift
func solution(_ num:Int) -> Int {
    var count = 0
    var temp = num
    while(temp != 1){
        count += 1
        if temp % 2 == 0 { temp = temp/2}
        else { temp = temp*3 + 1}
        
        if count == 500 {
            return -1
        }
    }
    return count
}